testing/mod.ts

```diff
@@ -140,49 +141,79 @@ async function createTestCase(
     }
   }
   if (previousPrinted(name, results)) {
-    report(result);
+    report(result, consoleLog);
   }
 }
 
 function initTestCases(
   stats: TestStats,
   results: TestResults,
   tests: TestDefinition[],
-  exitOnFail: boolean
+  exitOnFail: boolean,
+  consoleLog: logginFunctionType
 ): Array<Promise<void>> {
-  return tests.map(createTestCase.bind(null, stats, results, exitOnFail));
+  return tests.map(
+    createTestCase.bind(null, stats, results, exitOnFail, consoleLog)
+  );
 }
 
 async function runTestsParallel(
   stats: TestStats,
   results: TestResults,
   tests: TestDefinition[],
+  disableLog: boolean,
   exitOnFail: boolean
 ): Promise<void> {
+  let l, e;
   try {
-    await Promise.all(initTestCases(stats, results, tests, exitOnFail));
+    if (disableLog) {
+      l = console.log;
+      console.log = function() {};
```

Comment:
@zekth I think you might have missed `e = console.error`

Reply:
Also consider replacing `function() {}` with single `noop`

Reply:
you right